pkgs/by-name/so/sonarlint-ls/package.nix

```diff
@@ -65,11 +59,11 @@ maven.buildMavenPackage rec {
   nativeBuildInputs = [ makeWrapper ];
 
   passthru = {
-    tests = {
-      sonarlint-ls-starts-successfully = runCommand "${pname}-test" { } ''
-        ${sonarlint-ls}/bin/sonarlint-ls > $out
-        cat $out | grep "SonarLint backend started"
-      '';
+
+    tests.version = testers.testVersion {
+      package = sonarlint-ls;
+      command = "sonarlint-ls -V";
+      version = builtins.substring 0 4 version;
```

Comment:
`versionCheckHook` is preferred to testers.testVersion, I think. `lib.versions.majorMinor version` would be better to handle minor versions > 99, and makes it easier to follow.

Reply:
The `lib.versions.majorMinor version` is indeed much clearer. Thanks for that.

> versionCheckHook is preferred to testers.testVersion, I think

The idea here is not to fail the build if the version is wrong but to check if I wrapped everything correctly and that the final wrapper actually starts. So I use it like an end to end test for the final wrapped binary. For this, I think that the `testers.testVersion` is more suitable. What do you think?

Reply:
If you add `nativeInstallCheckInputs = [ versionCheckHook ];`, `versionCheckProgramArg = "-V";` and `doInstallCheck = true;` it will do exactly that.

The "downside" is that a failing `sonarlint-ls -V` would fail the build. But in my opinion that would be better than shipping a broken package to users.

Reply:
Though I am not sure this will allow you to override the expected version

Reply:
@FliegendeWurst But is the `versionCheckHook` not run during the preInstallPhase and thus it is _only_ checking the upstream binary?

I don't want to test the raw upstream binary, that is their job. I want to test if I correctly wrapped and linked it with nixpkgs. 

I know that the `testers.testVersion` is not run during the build but only by the update script and I am fine with that.

Reply:
The hook runs during installCheck, after install.
To override the expected version I would suggest `preVersionCheck = "export version=${lib.versions.majorMinor version}";